### ISSUE TITLE
`yarn type` fails @pothos/core 3.31.0 or higher

### ISSUE THREAD

**Issue by 6br** [Cauen/prisma-generator-pothos-codegen#57]
Hello, thank you for developing this library.
I get a type error with `pothos/core` version 3.31.0 or higher. If there are type errors, I am unable to build TypeScript codes using `tsc`. I'd be glad if this issue is fixed.

```bash
cd examples/input-simple-sqlite/
yarn upgrade     # @pothos/core is updated to >= 3.31.0
yarn type            # it fails.
```

**Comment by Cauen** [comment]
Hi @6br 
Currently, the latest version of @pothos/core supported is 3.23.0
I should investigate further how to update to the latest version when I have some time.

**Comment by bmayen** [comment]
Is there specific functionality in pothos >3.23.0 that breaks this? The peerDeps in this lib seem overly restrictive since pothos ^3.x semver range *shouldn't* be introducing breaking changes. Of course, I say "shouldn't" with a grain of salt.

**Comment by Cauen** [comment]
Hi @bmayen 
Today we are directly dependent on the implementation of Pothos (even if the new version dont have breaking changes)
Maybe things can change in the future due to updates here: https://github.com/Cauen/prisma-generator-pothos-codegen/issues/39

The type error starts at 3.31.0
I've opened an issue to discuss further the possible solution
https://github.com/hayes/pothos/issues/1111

**Comment by Cauen** [comment]
Hi @bmayen @6br 
Due to the latest update in @pothos/core i was able to fix this issue in new the version 0.6.4
Can you guys try this new version? :)

**Comment by bmayen** [comment]
Great! It might take me a few days to get to this due to the holidays, but I'll let you know as soon as I do!
